hddm/cnn/config.py

```diff
@@ -0,0 +1,274 @@
+
+
+# ddm_rel_checkpoint_dict = dict('model_checkpoint_path:': "/ddm_210500.ckpt-210500")
+
+# for key, val in ddm_rel_checkpoint_dict:
+#     ddm_rel_checkpint_dict[key] = hddm.__path__[0] + ddm_rel_checkpint_dict[val]
+
+class Config(object):
+
+    def __init__(self, model=None, bins=None, N=None):
+        # Directory setup
+        self.base_dir = hddm.__path__[0]  #+ '/Users/afengler/OneDrive/git_repos/hddm/hddm/'
+        self.data_dir = 'data'
+        self.tfrecord_dir = 'tfrecords'
+        self.summary_dir = 'summaries'
+
+        # training dataset features
+        self.isBinned = True
+        if model is None:
+            model = 'ddm_sdv'
+        if bins != None:
+            self.nBins = bins
+        else:
+            self.nBins = 512
+        self.nDatapoints = 100000
+
+        if N != None:
+            self.N = N
+        else:
+            self.N = 1024
+
+        self.method_options = {'ddm': self.ddm_initialize, 'angle': self.angle_initialize, 'weibull_cdf': self.weibull_initialize, 'ornstein': self.ornstein_initialize,
+                               'full_ddm': self.full_ddm_initialize, 'race_model_3': self.race_model_3_initialize, 'race_model_4': self.race_model_4_initialize,'race_model_6':self.race_model_6_initialize,
+                               'lca_3': self.lca_3_initialize, 'lca_4': self.lca_4_initialize,
+                               'ddm_seq2': self.ddm_seq2_initialize,
+                               'ddm_par2': self.ddm_par2_initialize,
+                               'ddm_mic2': self.ddm_mic2_initialize,
+                               'levy': self.levy_initialize,
+                               'full_ddm2': self.full_ddm2_initialize,
+                               'ddm_sdv': self.ddm_sdv_initialize}
+
+        # select model
+        self.method_options[model](self.nBins)
+
+        # select dataset
+        param_recovery_folder = os.path.join('../data/{}'.format(self.model_name),
+                                        'parameter_recovery_data_binned_{}_nbins_{}_n_{}'.format(int(self.isBinned), self.nBins, self.N))
+
+        self.inference_dataset = glob.glob('{}/*'.format(param_recovery_folder))
+        
+        self.dataset_dir = os.path.join(self.model_name, 'training_data_binned_{}_nbins_{}_n_{}'.format(int(self.isBinned),self.nBins,self.nDatapoints))
+        self.refname = self.dataset_dir.replace('/','_')
+
+        self.train_tfrecords = self.refname+'_train.tfrecords'
+        self.val_tfrecords = self.refname+'_val.tfrecords'
+        self.test_tfrecords = self.refname+'_test.tfrecords'
+
+        """
+        Specify whether or not to treat gaussian errors as isotropic
+        This only matters for the "reverse" model
+        """
+        self.full_cov_matrix = False
+        self.model_suffix = 'full_cov' if self.full_cov_matrix else 'isotropic'
+
+        # Data configuration
+        self.results_dir = '/media/data_cifs/projects/prj_approx-bayes/projectABC/results/'
```

Comment:
Why are these hardcoded?

Reply:
Ah no that's right, these paths are actually all irrelevant. 
As a global comment on the cnn code --> the cnn part is not emphasized because it has performance issues relative to the mlp stuff. The code wasn't fully cleaned up, as I realize now.

Reply:
I would add only the minimum that is necessary here, and only the best bits, the more code there is to maintain, the more headache.

Reply:
severely reduced the cnn related code